src/hotspot/share/classfile/classLoaderData.cpp

```diff
@@ -930,20 +930,56 @@ void ClassLoaderData::print_value_on(outputStream* out) const {
 void ClassLoaderData::print_value() const { print_value_on(tty); }
 
 #ifndef PRODUCT
-void ClassLoaderData::print_on(outputStream* out) const {
-  out->print("ClassLoaderData CLD: " PTR_FORMAT ", loader: " PTR_FORMAT ", loader_klass: %s {",
-              p2i(this), p2i(_class_loader.ptr_raw()), loader_name_and_id());
-  if (has_class_mirror_holder()) out->print(" has a class holder");
-  if (claimed()) out->print(" claimed");
-  if (is_unloading()) out->print(" unloading");
-  out->print(" metaspace: " INTPTR_FORMAT, p2i(metaspace_or_null()));
+class PrintKlassClosure: public KlassClosure {
+  outputStream* _out;
+public:
+  PrintKlassClosure(outputStream* out): _out(out) { }
 
+  void do_klass(Klass* k) {
+    ResourceMark rm;
+    _out->print("%s,", k->external_name());
+  }
+};
+void ClassLoaderData::print_on(outputStream* out) const {
+  ResourceMark rm;
+  out->print_cr("ClassLoaderData(" PTR_FORMAT ")", p2i(this));
+  out->print_cr(" - name                %s", loader_name_and_id());
+  if (!_holder.is_null()) {
+    out->print   (" - holder              ");
+    _holder.print_on(out);
+    out->print_cr("");
+  }
+  out->print_cr(" - class loader        %p", _class_loader.ptr_raw());
```

Comment:
We don't use  the %p in VM shared code as it behaves differently across platforms - use PTR_FORMAT.